src/Columns/ColumnsCommon.h

```diff
@@ -15,6 +20,40 @@ namespace ErrorCodes
     extern const int LOGICAL_ERROR;
 }
 
+/// Transform 64-byte mask to 64-bit mask
+inline UInt64 Bytes64MaskToBits64Mask(const UInt8 * bytes64)
+{
+#if defined(__AVX512F__) && defined(__AVX512BW__)
+    static const __m512i zero64 = _mm512_setzero_epi32();
+    UInt64 res = _mm512_cmp_epi8_mask(_mm512_loadu_si512(reinterpret_cast<const __m512i *>(bytes64)), zero64, _MM_CMPINT_EQ);
+#elif defined(__AVX__) && defined(__AVX2__)
+    static const __m256i zero32 = _mm256_setzero_si256();
+    UInt64 res =
+        (static_cast<UInt64>(_mm256_movemask_epi8(_mm256_cmpeq_epi8(
+        _mm256_loadu_si256(reinterpret_cast<const __m256i *>(bytes64)), zero32))) & 0xffffffff)
+        | (static_cast<UInt64>(_mm256_movemask_epi8(_mm256_cmpeq_epi8(
+        _mm256_loadu_si256(reinterpret_cast<const __m256i *>(bytes64+32)), zero32))) << 32);
+#elif defined(__SSE2__) && defined(__POPCNT__)
+    static const __m128i zero16 = _mm_setzero_si128();
+    UInt64 res =
+        (static_cast<UInt64>(_mm_movemask_epi8(_mm_cmpeq_epi8(
+        _mm_loadu_si128(reinterpret_cast<const __m128i *>(bytes64)), zero16))) & 0xffff)
+        | ((static_cast<UInt64>(_mm_movemask_epi8(_mm_cmpeq_epi8(
+        _mm_loadu_si128(reinterpret_cast<const __m128i *>(bytes64 + 16)), zero16))) << 16) & 0xffff0000)
+        | ((static_cast<UInt64>(_mm_movemask_epi8(_mm_cmpeq_epi8(
+        _mm_loadu_si128(reinterpret_cast<const __m128i *>(bytes64 + 32)), zero16))) << 32) & 0xffff00000000)
+        | ((static_cast<UInt64>(_mm_movemask_epi8(_mm_cmpeq_epi8(
+        _mm_loadu_si128(reinterpret_cast<const __m128i *>(bytes64 + 48)), zero16))) << 48) & 0xffff000000000000);
+#else
+    UInt64 res = 0;
+    const UInt8 * pos = bytes64;
+    const UInt8 * end = pos + 64;
+    for (; pos < end; ++pos)
+        res |= ((*pos == 0)<<(pos-bytes64));
```

Comment:
Unfortunately, this code is wrong and it breaks all non-x86 builds.

Reply:
#31574

Reply:
Noticed. thanks for info.